pkg/reconciler/common/extensions.go

```diff
@@ -39,6 +40,9 @@ func NoExtension(context.Context) Extension {
 
 type nilExtension struct{}
 
+func (nilExtension) Manifests() []mf.Manifest {
```

Comment:
Should this take the KComponent to be able to dynamically decide based on Spec settings?

Reply:
Yes good idea does not hurt anything to add it.